src/main/java/com/tabgok/entity/Machine.java

```diff
@@ -0,0 +1,54 @@
+package com.tabgok.entity;
+
+import com.tabgok.data.StringData;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * The machine is a model of a machine.  Importantly, the machine is the top-level
+ * model for a machine, meaning it contains within it either references to, or values of,
+ * everything within a machine from Hardware to Filesystems to applications.
+ */
+public class Machine extends Entity {
+    private StringData MachineID;
```

Comment:
I'm not picking and choosing fields for what is immutable and what isn't, maybe I should.  For instance, if a disk gets moved in/out of a machine, I'd expect the values to update accordingly -> but then again, I'd want a new disk and not the old one.

The machine is difficult though, because the ID is read from disk (there's a linux file for ID) so I can't set it in the constructor easily, at the moment.